src/components/Dialogs/PeriodDimension/PeriodDimension.js

```diff
@@ -118,18 +120,23 @@ export const PeriodDimension = ({ dimension, onClose }) => {
             : OPTION_PRESETS
     )
 
+    const inputType = useSelector(sGetUiInputType)
+
+    const { programId } = extractDimensionIdParts(dimension.id, inputType)
+
     const updatePeriodDimensionItems = (items) => {
         const { uiItems, metadata } = items.reduce(
             (acc, item) => {
-                acc.uiItems.push(item.id)
+                const itemId = programId ? `${programId}.${item.id}` : item.id
```

Comment:
This should use the utils instead to make sure program id is only added when applicable (i.e. when input === tracked entity)